Milestone4/logic

YoseptF opened the pull request:
I added the logic to run the calculator
- The operator module uses big.js to run math operations
- The calculate module uses operator to modify the state of the application

I also modified some of the existing logic in the app to make it easier to use these 2 new modules.

YoseptF committed:
I created the first layout and the components to be used in the app

YoseptF committed:
I made the layout for the claculator :D, also now it's separated into props that communicate between themselves

YoseptF committed:
Merge branch 'development' into milestone2/basicFeatures

YoseptF committed:
I changed some names in the Display component to comply with the milestone requirements
Also I made a dry-er version of the button panel

YoseptF committed:
I changed the name 'value' to 'result', and made it a string in order to comply with the milestone requirements
I also made a dry-er version of the button panel
Finally I made some changes in the way the windows recives an array

YoseptF committed:
The result prop now default to 0

YoseptF committed:
Merge branch 'milestone2/basicFeatures' of github.com:YoseptF/react-calculator into milestone3/styling

YoseptF committed:
I modified the readme file to look more professional

YoseptF committed:
Merge branch 'milestone2/basicFeatures' of github.com:YoseptF/react-calculator into milestone3/styling

YoseptF committed:
I added the required styling from the microverse milestone 3

YoseptF committed:
I added the operate and calculate modules :d

bellom commented on src/logic/operate.js:
```diff
@@ -0,0 +1,23 @@
+  const bigTwo = Big(numberTwo);
+  switch (operation) {
+    case '-':
+      return bigOne - bigTwo;
```
Good job getting this far.
The way you used the new `big.js` I am curious about it.
kindly check out what we have on the doc.
`https://mikemcl.github.io/big.js/`
Example:
`bigOne.minus(bigTwo)`

bellom commented on src/logic/operate.js:
```diff
@@ -0,0 +1,23 @@
+    case 'X':
+      return bigOne * bigTwo;
+    case '%':
+      return bigOne % bigTwo;
```
Note: case `%` require just the first digit which is `bigOne`

bellom commented on src/logic/calculate.js:
```diff
@@ -0,0 +1,39 @@
+    case '6':
+    case '7':
+    case '8':
+    case '9':
+      newData.next += buttonName;
+      break;
+
+    case '+/-':
+      newData.total *= -1;
+      newData.next *= -1;
+      break;
+
+    case 'AC':
+      newData.total = '0';
+      newData.next = '0';
+      newData.operation = null;
+      break;
+
+    default:
+      newData.total = operate(...dataObject);
+      newData.next = '0';
+      newData.operation = null;
+      break;
+  }
+  return newData;
```
we need to add more conditions, like;
1. division zero error handling(if we divide a digit by zero we should get an error message)
2. when `.` return `0.`

bellom reviewed (changes requested):
## Hi @YoseptF  :clap: 

### Awesome job so far!! :+1: 

**NOTE: kindly fix the linter's error**
fix all errors:** checkout this [linters-config#troubleshooting](https://github.com/microverseinc/linters-config#troubleshooting) for help. Example below;
![image](https://user-images.githubusercontent.com/31897434/63073279-0e393680-bf20-11e9-9c84-f142caf1e675.png)
#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

YoseptF committed:
I resolved the linter errors, and added 2 new clauses to the division case
I also aplyied correctly the big.js operations and moved the percentaje to the calculate module

YoseptF committed:
I added 1 more line to comply with stickler

AdnanAfsari reviewed (approved):
**Status: Approved**

Hello @YoseptF ✋ 

Nice work, You have made changes requested by the previous reviewer.It's a 👍 .

Happy Coding
Adnan Afsari
